src/toil/lib/ec2nodes.py

```diff
@@ -192,11 +197,14 @@ def fetchEC2InstanceDict(awsBillingJson: Dict[str, Any], region: str) -> Dict[st
             normal_use = i.get('usagetype').endswith('BoxUsage:' + i['instanceType'])  # not reserved or unused
             if normal_use:
                 disks, disk_capacity = parseStorage(v["attributes"]["storage"])
+                architecture = 'arm64' if re.search(".*\dg.*\..*", i["instanceType"]) else 'amd64'
```

Comment:
Oops, I was wrong, you do have the code to determine the architecture, right here.